cou/steps/analyze.py

```diff
@@ -78,6 +79,30 @@ def _split_control_plane_and_data_plane(
                 control_plane.append(app)
         return control_plane, data_plane
 
+    @classmethod
+    def _remove_data_plane_apps(
+        cls, control_plane: list[OpenStackApplication], data_plane: list[OpenStackApplication]
+    ) -> None:
+        """Move control plane applications to data plane if any of its units is on data plane.
+
+        :param control_plane: List of control plane applications.
+        :type control_plane:  list[OpenStackApplication]
+        :param data_plane: List of data plane applications.
+        :type data_plane:  list[OpenStackApplication]
+        """
+        data_plane_machines = set().union(
+            *[{unit.machine for unit in app.units} for app in data_plane]
+        )
```

Comment:
```suggestion
        data_plane_machines = {unit.machine for unit in app.units for app in data_plane}
```

Reply:
this does not work

Reply:
```suggestion
        data_plane_machines = {unit.machine for app in data_plane for unit in app.units}
```
Sorry wrong order. Here is an example how you can do it in Python
```python
[f"{i}" for app in [[1,2,3], [3,4,5]] for i in app]
```

Reply:
ok